File: toolloop-back/src/main/java/com/toolloop/model/dto/UpdateToolRequest.java
package com.toolloop.model.dto;

import io.quarkus.runtime.annotations.RegisterForReflection;

import java.math.BigDecimal;
import java.util.List;

@RegisterForReflection
public record UpdateToolRequest(
    String name,
    String description,
    BigDecimal pricePerDay,
    BigDecimal securityDeposit,
    Long categoryId,
    String condition,
    List<String> photoKeys,
    ToolAvailabilityDTO availability
) { }

File: toolloop-front/src/app/core/models/dto/update-tool-request.ts
import { ToolAvailabilityDTO } from './tool-availability-dto';

export interface UpdateToolRequest {
    name: string;
    description: string;
    pricePerDay: number;
    securityDeposit: number;
    categoryId: number;
    condition: string;
    availability: ToolAvailabilityDTO;
}
